Automattic/wp-calypso
# Modernize wpcom-proxy-request

This PR modernizes `wpcom-proxy-request` in several ways:
- Switches to building through `calypso-build`
- Updates `debug` dependency
- Reorganizes both source and package directories to match `calypso-build` expectations

It also switches over to using this package in Calypso and `package/data-stores`. It keeps the package private in the meantime, though, in case we want to make further changes before the next major version release as an independent npm package.

This PR should remove the last dependency on `babel-runtime` (as opposed to the newer `@babel/runtime`, which we already use), and it should relegate usage of `core-js@2` to devdocs, leaving the main entry points with just `core-js@3`.

The size savings are smaller than I hoped for (~7KB compressed), but still very desirable.

#### Changes proposed in this Pull Request

* Modernize `wpcom-proxy-request`
* Switch to the in-repo package in Calypso and `package/data-stores`
* Fix some `wpcom-proxy-request` lint errors

#### Testing instructions

Pretty much all API requests go through `wpcom-proxy-request`, so some casual testing of the live branch should be enough to ensure that nothing went horribly wrong with this change.
